Lib/test/test_tempfile.py

```diff
@@ -979,6 +979,88 @@ def test_del_on_close(self):
         finally:
             os.rmdir(dir)
 
+    def test_not_del_on_close_if_delete_on_close_false(self):
+        # A NamedTemporaryFile is NOT deleted when closed if
+        # delete_on_close= False, but is deleted on content manager exit
+        dir = tempfile.mkdtemp()
+        try:
+            with tempfile.NamedTemporaryFile(dir=dir,
+                                             delete=True,
+                                             delete_on_close=False) as f:
+                f.write(b'blat')
+                f_name = f.name
+                f.close()
+                with self.subTest():
+                    # Testing that file is not deleted on close
+                    self.assertTrue(os.path.exists(f.name),
+                            "NamedTemporaryFile %s is incorrectly deleted\
+                             on closure when delete_on_close= False" % f_name)
```

Comment:
```suggestion
                    self.assertTrue(os.path.exists(f.name),
                            f"NamedTemporaryFile {f.name!r} is incorrectly deleted "
                            f"on closure when delete_on_close=False")
```
I switched to f-strings to fix the output message, `f.name` (instead of `f_name`) for consistency, and added `!r`.